### PR TITLE
Blubber Config + Docker setup

### PR DESCRIPTION
**Pull request by nikkhn**
This is an initial go at a blubber config and docker-compose setup. I am by no means an expert so feel free to poke/pry. I also removed a few things that I believe are taken care of by eslint-config-wikimedia npm module.

### CODE REVIEW

**Comment by niedzielski** [comment]
@nikkhn, I'm guessing it's ok to reopen this.

File: .stylelintignore [hunk 1]
```diff
@@ -3,6 +3,8 @@
 /docs/
 
 *.json
+*.yaml
+Dockerfile
```
Comment (niedzielski) [inline .stylelintignore:7]: Would you mind using an anchored path for the Dockerfile? `/Dockerfile`. It's definitely not a big deal but the rationale is that we've been erring on side of precision in trying to keep our ignore files very intentional and specific.

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,8 @@
+FROM docker-registry.wikimedia.org/nodejs10-devel
+RUN apt-get update && \
+	apt-get install -y \
+		build-essential \
+		# ca-certificates \
+		python-pkgconfig \
+		git
+RUN npm install
```
Comment (niedzielski) [inline Dockerfile:8]: Is this the recommended command from RelEng / WMDE or should we use `npm ci`?

File: .pipeline/blubber.yaml [hunk 3]
```diff
@@ -0,0 +1,17 @@
+version: v4
```
Comment (niedzielski) [inline .pipeline/blubber.yaml:1]: Please add a new section to the readme for Docker. The section should cover at least what Docker is used for in this repo and how to maintain it. For example, if adding a dependency now requires rebuiding the Docker image each time, the readme should cover that. If Docker is optional for local development, which I believe it is, the section should explicit call that out.

**Comment by nikkhn** [comment]
@niedzielski I actually think I will just cherry-pick my commits into the gerrit repo like you suggested today. (Was trying to be slightly lazy and just keep this branch open and merge it in github haha)

File: .pipeline/blubber.yaml [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+    build:
+        base: docker-registry.wikimedia.org/nodejs-devel
+        copies: [local]
+        apt:
+            packages: [git, build-essential, python-pkgconfig] # needed for npm install
+        node:
+            requirements: [package.json, package-lock.json]
+            env: development
+        builder:
+            command: [npm, build]
```
Comment (niedzielski) [inline .pipeline/blubber.yaml:14]: Does this line run `npm build` or `npm run build`? I thought we wanted the latter but could be mistaken.

File: package.json [hunk 5]
```diff
@@ -46,7 +46,7 @@
     "format:js": "npm -s run linter:js -- --fix .",
     "format:styles": "npm -s run linter:styles -- --fix .",
     "linter:etc": "prettier .",
-    "linter:js": "eslint --cache --max-warnings 0 --report-unused-disable-directives --ext .js,.json,.ts,.vue",
```
Comment (niedzielski) [inline package.json:49]: Can you tell me what this change is for?

File: package-lock.json [hunk 6]
```diff
@@ -6260,6 +6260,16 @@
         "del": "^4.1.1"
```
Comment (niedzielski) [inline package-lock.json:6260]: From what I see, there are no changes to dependencies in this PR so this file should not be changed.

File: docker-compose.yaml [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+            dockerfile: ./Dockerfile
+        image: wikimedia/wvui
+        volumes:
+            - ./:/app
+        working_dir: /app
+    storybook:
+        extends:
+            service: node
+        ports:
+            - "3003:3003"
```
Comment (niedzielski) [inline docker-compose.yaml:15]: Is this to match Storybook? If so, please add a comment.

**Review by niedzielski — CHANGES_REQUESTED** [review]

**Comment by niedzielski** [comment]
@nikkhn, just saw your comment. Works for me :+1:

**Comment by nikkhn** [comment]
@niedzielski I will copy over your pr comments to gerrit and address there so they are documented

**Pull request closed without merging**